File: Recursion_Array/lastIndex.cpp
#include<iostream>
#include<vector>
using namespace std;

int last(vector<int>& arr, int idx, int data)
{
    if(idx == -1)
    {
        return -1;
    }
    if(arr[idx] == data)
    {
        return idx;
    }
    return last(arr, idx - 1, data);
}

void input(vector<int>& arr)
{
    for(int i = 0; i < arr.size(); i++)
    {
        cin >> arr[i];
    }
}

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    input(arr);
    int data;
    cin >> data;
    cout << last(arr, n -1, data);
}

File: Recursion_Array/lastIndex.java
import java.util.Scanner;
public class lastIndex
{
    public static Scanner scn = new Scanner(System.in);
    public static int last(int[] arr, int idx, int data)
    {
        if(idx == -1)
        {
            return -1;
        }
        if(arr[idx] == data)
        {
            return idx;
        }
        return last(arr, idx - 1, data);
    }

    public static void input(int[] arr)
    {
        for(int i = 0; i < arr.length; i++)
        {
            arr[i] = scn.nextInt();
        }
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] arr = new int[n];
        input(arr);
        int data = scn.nextInt();
        System.out.println(last(arr, n - 1, data));
    }
}